### DISCUSSION TITLE
create column

### DISCUSSION THREAD

**Discussion by mortezabonak** [adonisjs/core#2510] [Help]
hi
how can i create column in migration file using raw query?
i want to create column in integer array type but my query does not apply.
this is my migration file:
```
import BaseSchema from '@ioc:Adonis/Lucid/Schema'


export default class Products extends BaseSchema {
  protected tableName = 'products'

  public async up() {
    this.raw(
      'ALTER TABLE products ADD stages integer ARRAY'
    )

  }

  public async down() {
    this.raw(
      'ALTER TABLE products DROP COLUMN stages'
    )
  }
}
```

**Comment by thetutlage** [comment]
Need more information to provide any help. Do you receive any error? Also share the actual outcome and what is expected outcome.

**Reply by mortezabonak** [reply]
there was no error log but the column was not created.
so i used `table.specificType('stages', 'INT[]')` in migration file and worked as well.

**Reply by thetutlage** [reply]
I see, it should be

```ts
this.schema.raw()
```
